pkg/manifest/simple.go

```diff
@@ -106,9 +106,9 @@ func (m *simpleManifest) Store(ctx context.Context, mode storage.ModePut) (swarm
 		return swarm.ZeroAddress, fmt.Errorf("manifest marshal error: %w", err)
 	}
 
-	sp := splitter.NewSimpleSplitter(m.storer, mode)
-
-	address, err := file.SplitWriteAll(ctx, sp, bytes.NewReader(data), int64(len(data)), m.encrypted)
+	pipe := pipeline.NewPipeline(ctx, m.storer, mode)
```

Comment:
on a sidenote this is not so great that we have this dependency buried here. the ability to marshal a manifest should be possible from the calling context

Reply:
That will be problem for new 'mantaray' implemetation, which uses some internal `LoadSaver` interface, which lazily saves and loads required Trie nodes, to optimize space/time.